### ISSUE TITLE
Make option to check parsing only after user starts a new line. 

### ISSUE THREAD

**Issue by Feniks-Gaming** [godotengine/godot#26797]
**Godot version:**
3.1

**OS/device including version:**
Windows 10

**Issue description:**

Currently if you stop mid way typing a line of code Godot automatically checks if this line has an error. As a line is not finished yet Godot automatically reports the error. This is undesired behaviour because user is still working on that line of code and is well aware this line is incomplete.  We should have option for Godot to check parsing only after user moves to new line. As long as cursor is on the same line Godot should ignore checking this line for errors. 

**Steps to reproduce:**

Type any line of code and don't finish it.

**Minimal reproduction project:**
N?A

**Comment by Paulb23** [comment]
Related / Duplicate of #7502

**Comment by akien-mga** [comment]
The lobby is stronger on this one, so I'll close the former one as duplicate.

**Comment by DriNeo** [comment]
There is a little problem with this behaviour. The line may contain a wrong subexpression. So the line is coloured for an error before the end. 
I guess that can be annoying if after writing a long "if" test, as example, you discover the error once you end your typing.
In #7502 I suggested to simply change the error colouring style in order to not hide the line. I guess the point of this current issue is to improve legibility.

**Comment by Two-Tone** [comment]
> I guess that can be annoying if after writing a long "if" test, as example, you discover the error once you end your typing.

The issue is that it'll error anyways because the hypothetical if statement requirements are not complete. Waiting til the user leaves the line is the most logical solution as they could still be writing it at any point. You'll get substantially less false positives that way.

**Comment by Calinou** [comment]
Feature and improvement proposals for the Godot Engine are now being discussed and reviewed in a dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, we are now closing all older feature proposals on the main issue tracker.

If you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal). Thanks in advance!
